extensions/notebook/src/test/common/utils.test.ts

```diff
@@ -134,7 +137,169 @@ describe('Utils Tests', function () {
 		});
 	});
 
+	describe('executeBufferedCommand', () => {
+
+		it('runs successfully', async () => {
+			await utils.executeBufferedCommand('echo hello', {}, new MockOutputChannel());
+		});
+
+		it('errors correctly with invalid command', async () => {
+			await should(utils.executeBufferedCommand('invalidcommand', {}, new MockOutputChannel())).be.rejected();
+		});
+	});
+
+	describe('executeStreamedCommand', () => {
+
+		it('runs successfully', async () => {
+			await utils.executeStreamedCommand('echo hello', {}, new MockOutputChannel());
+		});
+
+		it('errors correctly with invalid command', async () => {
+			await should(utils.executeStreamedCommand('invalidcommand', {}, new MockOutputChannel())).be.rejected();
+		});
+	});
+
+	describe('isEditorTitleFree', () => {
+		it('title is free', () => {
+			should(utils.isEditorTitleFree('MyTitle')).be.true();
+		});
+
+		it('title is not free with text document sharing name', async () => {
+			await vscode.workspace.openTextDocument();
+			should(utils.isEditorTitleFree('Untitled-1')).be.false();
+		});
+
+		it('title is not free with notebook document sharing name', async () => {
+			await azdata.nb.showNotebookDocument(vscode.Uri.parse(`untitled:Untitled-1`));
```

Comment:
would love to also test the notebook naming convention

e.g. Notebook-0, Notebook-1, etc.

Reply:
Good call - so right now the Notebook-* naming convention is only exposed through an internal command (an extension would have to specify an untitled editor with that name itself). I added a test for it anyways and doing so found out some wrong-ish behavior where we weren't awaiting the creation of the editor correctly when executing the command.

I fixed this but it shouldn't have any actual behavior change. I did change it so that it will properly reject with an error if the creation fails instead of just displaying the error (it's better behavior IMO to let the caller handle errors). 

I did test it and verified that the action does still properly display the error when it fails without having the extension do it itself.